### PR TITLE
feat(manage): add memory notes management to game resources

### PR DESCRIPTION
**Pull request by wescopeland**
This PR adds another relation manager to the management app's game resource. Now, developers can manage code notes from the game resource's page, similar to achievements and hashes. The intention is to fully mirror the code note management behavior in the player-facing app, and in a subsequent release have the functionality live exclusively in the management app.

There are a few minor behavior differences:
* Only 200 code notes are loaded per page. This solves the Dragon Quest VIII problem where that game's code notes page will not load locally.
* Users can now search for code notes via the `body` attribute.
* Optionally, users can view a column for the `updated_at` value.
* Users can filter by note author.

Screenshot 2024-07-21 at 4 36 40 PM

Screenshot 2024-07-21 at 4 36 45 PM

Screenshot 2024-07-21 at 4 36 52 PM

Screenshot 2024-07-21 at 4 36 59 PM

Screenshot 2024-07-21 at 4 37 07 PM

Screenshot 2024-07-21 at 4 37 11 PM

### CODE REVIEW

File: app/Filament/Resources/GameResource/RelationManagers/MemoryNotesRelationManager.php [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+        return $table
+            ->recordTitleAttribute('address')
+            ->searchPlaceholder('Search (Body)')
+            ->modifyQueryUsing(fn (Builder $query) => $query->where('body', '!=', ''))
+            ->columns([
+                Tables\Columns\TextColumn::make('address_hex')
+                    ->label('Address')
+                    ->fontFamily(FontFamily::Mono)
+                    ->color('info'), // Add some visual distinction between the address and the body.
```
Comment (wescopeland) [inline app/Filament/Resources/GameResource/RelationManagers/MemoryNotesRelationManager.php:54]: Screenshot 2024-07-21 at 4 39 13 PM

I found myself getting disoriented when scrolling through long lists of notes. Changing the address color helped the problem a lot.

File: app/Filament/Resources/GameResource/RelationManagers/MemoryNotesRelationManager.php [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+            ])
+            ->actions([
+                Tables\Actions\ActionGroup::make([
+                    Tables\Actions\EditAction::make()
+                        ->form([
+                            Forms\Components\Textarea::make('body')
+                                ->label('Body')
+                                ->autosize()
+                                ->required(),
+                            ])
```
Comment (wescopeland) [inline app/Filament/Resources/GameResource/RelationManagers/MemoryNotesRelationManager.php:89]: I confirmed that whitespace is respected in RALibretro from notes saved using this control.

File: app/Filament/Resources/GameResource/RelationManagers/MemoryNotesRelationManager.php [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+                        })
+                        ->visible(function (MemoryNote $memoryNote): bool {
+                            /** @var User $user */
+                            $user = auth()->user();
+
+                            return $user->can('delete', $memoryNote);
+                        }),
+                ]),
+            ])
+            ->paginated([200])
```
Comment (Jamiras) [inline app/Filament/Resources/GameResource/RelationManagers/MemoryNotesRelationManager.php:152]: 200 is a lot of items to scroll through. Maybe limit it to 100 or even 50?

File: app/Models/MemoryNote.php [hunk 4]
```diff
@@ -28,8 +34,25 @@ class MemoryNote extends BaseModel
...
 
+    // == logging
+
+    public function getActivitylogOptions(): LogOptions
+    {
+        return LogOptions::defaults()
+            ->logOnly([
+                'user_id',
+            ])
+            ->dontSubmitEmptyLogs();
```
Comment (Jamiras) [inline app/Models/MemoryNote.php:45]: Something about this is causing log entries to be generated that only contain the user_id when modifying code notes created by the user.
```
{"attributes":{"user_id":43495},"old":{"user_id":43495}}
```

File: app/Filament/Resources/GameResource/RelationManagers/MemoryNotesRelationManager.php (hunk 3 above)
Comment (wescopeland) [reply app/Filament/Resources/GameResource/RelationManagers/MemoryNotesRelationManager.php]: Sounds good to me. In latest I've set the default to 50 and optionally allow the user to bump up to 100.

File: app/Models/MemoryNote.php (hunk 4 above)
Comment (wescopeland) [reply app/Models/MemoryNote.php:46]: Good catch, fixed in latest with the addition of a `->logOnlyDirty()`.

**Review by Jamiras — APPROVED** [review]

**Pull request merged**